### PR TITLE
TOOLS-2708: Atlas recommended connection string for mongostat doesn't work

### PR DESCRIPTION
**Pull request by varsha804**
The connection string failure seems to be a result of [this](https://github.com/mongodb/mongo-tools/commit/2aa1d7079b20817d17649ff7b43b2d79b69f1c8e) code change. Since `opts.Port` and `opts.Host` are not set when a user only uses `--uri`, `fullhost` gets set to `localhost:27017`, thereby incorrectly rewriting the hostname to `localhost`.

Manual testing:
1. Atlas cluster
2. mlaunch cluster
3. `--discover` without a connection string
4. Connection string as a positional argument without `--uri`
Screen Shot 2020-09-15 at 2 41 38 PM

### CODE REVIEW

File: mongostat/mongostat.go
Comment (huan-Mongo) [inline mongostat/mongostat.go:247]: Since we are not replacing the `Hosts` in ConnString with single node `fullHost`, the above code 
```	newCS, err := rewriteURI(uriCopy.ConnectionString, fullHost)
	if err != nil {
		return nil, err
	}
```
doesn't make sense in a replica-set connection anymore. I guess it didn't affect the tests, but it might have some side impact. 

I suggest we put the `rewriteURI` in the below if condition as well.

**Review by huan-Mongo — CHANGES_REQUESTED** [review]

File: mongostat/mongostat.go
Comment (varsha804) [reply mongostat/mongostat.go:247]: Yeah, that's a good point!

**Review by tfogo — CHANGES_REQUESTED** [review]
> Since `opts.Port` and `opts.Host` are not set when a user only uses `--uri`

I think this is the underlying bug though. What you have works, but I think it's more of a band-aid. I think we should fix the problem in `ToolOptions.setOptionsFromURI()` instead.

**Comment by varsha804** [comment]
Updated manual testing:
Screen Shot 2020-09-23 at 2 46 08 PM

**Review by tfogo — CHANGES_REQUESTED** [review]

File: vendor/github.com/mongodb/mongo-tools-common/options/options.go
Comment (tfogo) [reply vendor/github.com/mongodb/mongo-tools-common/options/options.go:601]: Oh I didn't see your question about this on the mtc PR - I'll respond there.

**Review by tfogo — APPROVED** [review]
lgtm once the mtc PR is merged.

**Review by huan-Mongo — APPROVED** [review]
LGTM after revendor

**Pull request merged**
